themes/_core/js/cart.js

```diff
@@ -142,16 +142,23 @@ $(document).ready(() => {
 
     $.post(actionURL, query, null, 'json')
       .then((resp) => {
-        prestashop.emit('updateCart', {
-          reason: {
-            idProduct: resp.id_product,
-            idProductAttribute: resp.id_product_attribute,
-            idCustomization: resp.id_customization,
-            linkAction: 'add-to-cart',
-            cart: resp.cart,
-          },
-          resp,
-        });
+        if (!resp.hasError) {
+          prestashop.emit('updateCart', {
+            reason: {
+              idProduct: resp.id_product,
+              idProductAttribute: resp.id_product_attribute,
+              idCustomization: resp.id_customization,
+              linkAction: 'add-to-cart',
+              cart: resp.cart,
+            },
+            resp,
+          });
+        } else {
+          prestashop.emit('handleError', {
```

Comment:
```suggestion
          prestashop.emit('error', {
```

I guess this should be enough, it's not a handler at all

If you wan't to make it even more verbose, it could be `onError`, but it would be really duplicate with`prestashop.on('onError'...`

Reply:
Just used the current structure,
If need to change this, so other ones need to change,
We need a global name for cart errors, then handling based on event type in FO.

![Screenshot from 2022-05-24 17-18-30](https://user-images.githubusercontent.com/85633460/169984489-21e29d57-fe89-4cd2-8056-6f178d803cda.png)

Reply:
You are right, let's keep it as long as we use these core files